### DISCUSSION TITLE
QueryBuilder leftJoin with multiple conditions

### DISCUSSION THREAD

**Discussion by tolgap** [mikro-orm/mikro-orm#2535] [Q&A] [answered]
I'm using MikroORM v4.5.9

I'm trying to write a query with multiple conditions in `leftJoin`. The notation for `leftJoin` is:

```ts
QueryBuilder<ProductVariantEntity>.leftJoin(field: string, alias: string, cond?: any): QueryBuilder<ProductVariantEntity>
```

Because the `cond` is of type `any`, I'm not sure how to do this. Any pointers?

Quick example:
The query I have right now has 3 `where` conditions. I want to move the `"po.placedDate": { $gt: lastChangedDate }` and `"ps.remainingQuantity": { $gt: 0 }` conditions into their respective `leftJoin`, while also solving their foreign keys in the `leftJoin` as well. See query below:

```ts
    const qb = this.createQueryBuilder("pv");
    const query = qb
      .leftJoin("pv.pendingStocks", "ps")
      .leftJoin("ps.order", "po")
      .where({
        "po.placedDate": { $gt: lastChangedDate },
        "ps.remainingQuantity": { $gt: 0 },
        "pv.id": variantId,
      })
```

This is what I would like to achieve, if possible in some other way:

```ts
    const qb = this.createQueryBuilder("pv");
    const query = qb
      .leftJoin("pv.pendingStocks", "ps", { "pv.id": "ps.product_variant_id", "ps.remainingQuantity": { $gt: 0 } })
      .leftJoin("ps.order", "po", { "ps.order_id": "po.id", "po.placedDate": { $gt: lastChangedDate } })
      .where({
        "pv.id": variantId,
      })
```

**Comment by B4nan** [comment, marked as answer]
> This is what I would like to achieve, if possible in some other way:

And that code does not work? It is correct way to use the `cond` parameter. Not sure what other way would you want.

Here is a corresponding test:

https://github.com/mikro-orm/mikro-orm/blob/master/tests/QueryBuilder.test.ts#L367

**Reply by tolgap** [reply]
Thanks @B4nan . That test was exactly what I needed.

Apparently, I don't have to add the condition to resolve the foreign key. I was trying to manually do that myself 😓 . So the `cond?: any` parameter are _additional_ conditions 👍 .
